File: CreationalDesignPattern/BuilderPattern/python/Driver.py
from ConcreteBuilders.GamingComputerBuilder import GamingComputerBuilder
from ConcreteBuilders.OfficeComputerBuilder import OfficeComputerBuilder
from Director import Director


builder = GamingComputerBuilder()
computerDirector = Director(builder)
computerDirector.construct_computer()
computer = builder.get_computer()
print(computer)

builder = OfficeComputerBuilder()
director = Director(builder)
director.construct_computer()
print(builder.get_computer())

File: CreationalDesignPattern/BuilderPattern/java/Driver.java
package CreationalDesignPattern.BuilderPattern.java;

import CreationalDesignPattern.BuilderPattern.java.AbstractBuilders.ComputerBuilder;
import CreationalDesignPattern.BuilderPattern.java.ConcreteBuilders.GamingComputerBuilder;
import CreationalDesignPattern.BuilderPattern.java.ConcreteBuilders.OfficeComputerBuilder;
import CreationalDesignPattern.BuilderPattern.java.Product.Computer;

public class Driver {
    public static void main(String[] args){
        ComputerBuilder builder = new GamingComputerBuilder();
        Director computerDirector = new Director(builder);
        computerDirector.constructComputer();
        Computer gamingComputer = builder.getComputer();
        gamingComputer.printComputer();

        ComputerBuilder officeComputerBuilder = new OfficeComputerBuilder();
        Director officeComputerDirector = new Director(officeComputerBuilder);
        officeComputerDirector.constructComputer();
        Computer officeComputer = officeComputerBuilder.getComputer();
        officeComputer.printComputer();
    }
}
